sdk/trace/span.go

```diff
@@ -285,6 +290,7 @@ func (s *span) RecordError(err error, opts ...trace.EventOption) {
 	opts = append(opts, trace.WithAttributes(
 		semconv.ExceptionTypeKey.String(typeStr(err)),
 		semconv.ExceptionMessageKey.String(err.Error()),
+		semconv.ExceptionStacktraceKey.String(recordStackTrace(err)),
```

Comment:
Can we add an `EventOption` to specify if the use wants to include this or not? I would prefer to not include it by default.

Reply:
Not sure if I understand why this should be opt in? I feel stack trace is more like by default option and probably allowing them to opt in would be add more burden. Moreover, errors are unpredictable in nature and collecting extra information for root cause would be always helpful. As far as this change is not causing performance implication then I think by default stack trace capture would be a good option. However, I'm happy to hear your thoughts on why this should be opt-in :)

Reply:
Stack traces are larges amounts of data. We do not want to bloat telemetry with this data by default, this would impose additional transport and telemetry storage requirements on users. Additionally, this is the way it is specified: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/exceptions.md#attributes

Reply:
Okay. Makes sense! Yeah It would add overhead in terms of storage for sure. I can make this opt-in. IDK how to use `EventOptions` tho. Can you point me to some examples on how this can be implemented?

Reply:
Sure thing.

You will need to extend the `EventConfig` to include a new stack trace field (maybe `stackTrace bool`?).

https://github.com/open-telemetry/opentelemetry-go/blob/56c743ba7998613f046f51954b7ccf0f911cd29f/trace/config.go#L138-L152

That will require adding a getter method for what ever field you add.

Then you will want to add an explicit option that implements the `EventOption` interface for stacktraces. E.g.

```go

type stackTraceOption bool

func (o stackTraceOption) applyEvent(c *EventConfig)    { c.stackTrace = bool(o) }

func WithStackTrace(b bool) EventOption {
	return stackTraceOption(b)
}
```